src/raw_node.rs

```diff
@@ -145,11 +145,23 @@ impl Ready {
         if &ss != prev_ss {
             rd.ss = Some(ss);
         }
-        let hs = raft.hard_state();
+        let mut hs = raft.hard_state();
         if &hs != prev_hs {
             if hs.get_vote() != prev_hs.get_vote() || hs.get_term() != prev_hs.get_term() {
                 rd.must_sync = true;
             }
+            // If we hit a size limit when loading CommittedEntries, clamp
+            // our HardState.Commit to what we're actually returning. This is
+            // also used as our cursor to resume for the next Ready.
+            if let Some(committed_entries) = rd.committed_entries.as_ref() {
```

Comment:
What about `rd.committed_entries.as_ref().and_then(|c| c.last())`?

Reply:
OMG, it's really beautiful here to use the functional pattern.